pkg/client/unversioned/helper.go

```diff
@@ -468,12 +483,9 @@ func DefaultServerURL(host, prefix string, groupVersion unversioned.GroupVersion
 	}
 
 	// If the user specified a URL without a path component (http://server.com), automatically
-	// append the default prefix
+	// append the proxy prefix
 	if hostURL.Path == "" {
-		if prefix == "" {
-			prefix = "/"
-		}
-		hostURL.Path = prefix
+		hostURL.Path = path.Join(proxyPrefix, apiPath)
```

Comment:
right, so the plan is to NOT append it here. instead a "versionedAPIPath" will be an additional result returned by this func. hopefully the next refactoring I push makes more sense